buildflow/core/app/flow.py

```diff
@@ -197,6 +214,10 @@ def pipeline(
             )
             sink.enable_managed()
 
+        # Set up credentials
```

Comment:
I like it a lot more as a method! Would you mind updating the primtive_type thing above to use a method?

Reply:
done.